code/game/objects/items/weapons/grenades/flashbang.dm

```diff
@@ -5,120 +5,117 @@
 	origin_tech = list(TECH_MATERIAL = 2, TECH_COMBAT = 1)
 	var/banglet = 0
 
-	prime()
-		..()
-		for(var/obj/structure/closet/L in hear(7, get_turf(src)))
-			if(locate(/mob/living/carbon/, L))
-				for(var/mob/living/carbon/M in L)
-					bang(get_turf(src), M)
-
+/obj/item/weapon/grenade/flashbang/prime()
+	..()
+	for(var/obj/structure/closet/L in hear(7, get_turf(src)))
+		if(locate(/mob/living/carbon/, L))

```

Comment:
If we're cleaning up. Remove this first `if`. Iterating over an empty list will serve the exact same purpose.